functions/Copy-DbaQueryStoreConfig.ps1

```diff
@@ -38,120 +54,116 @@ You should have received a copy of the GNU General Public License along with thi
 
 Copy the Query Store configuration of the AdventureWorks database in the ServerA\SQL Instance and apply it to the WorldWideTraders database in the ServerB\SQL Instance.
 	
 #>
-	[CmdletBinding()]
+	[CmdletBinding(SupportsShouldProcess = $true)]
 	Param (
-		[parameter(Mandatory = $true, ValueFromPipeline = $true)][string[]]$SourceSqlServer,
-        [parameter(Mandatory = $true, ValueFromPipeline = $true)][string[]]$SourceDatabase,
-        [parameter(Mandatory = $true, ValueFromPipeline = $true)][string[]]$TargetSqlServer,
-		[PsCredential]$Credential
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object[]]$Source,
+		[System.Management.Automation.PSCredential]$SourceSqlCredential,
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object]$SourceDatabase,
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object[]]$Destination,
+		[System.Management.Automation.PSCredential]$DestinationSqlCredential,
+		[object[]]$DestinationDatabase,
+		[object[]]$Exclude,
+		[switch]$AllDatabases
 	)
-
-	DynamicParam {
-		if ($SourceSqlServer) {
-			return Get-ParamSqlDatabases -SqlServer $TargetSqlServer[0] -SqlCredential $Credential
-		}
-	}
-
+	
 	BEGIN
 	{
-		# Convert from RuntimeDefinedParameter object to regular array
-		$databases = $psboundparameters.Databases
-		$exclude = $psboundparameters.Exclude
-
-        Write-Verbose "Connecting to source: $SourceSqlServer"
-			try
-			{
-				$server = Connect-SqlServer -SqlServer $SourceSqlServer -SqlCredential $Credential
-
-			}
-			catch
-			{
-				if ($SourceSqlServer.count -eq 1)
-				{
-					throw $_
-				}
-				else
-				{
-					Write-Warning "Can't connect to $SourceSqlServer."
-					Continue
-				}
-			}
-
-        # Grab the Query Store configuration from the SourceDatabase through the Get-DbaQueryStoreConfig function
-        $SourceQSConfig = Get-DbaQueryStoreConfig -SqlServer $SourceSqlServer -Databases $SourceDatabase
-
+	
+		Write-Verbose "Connecting to source: $Source"
+		try
+		{
+			$sourceserver = Connect-SqlServer -SqlServer $Source -SqlCredential $SourceSqlCredential
+			
+		}
+		catch
+		{
+			Write-Warning "Can't connect to $Source."
+			continue
+		}
+		
+		# Grab the Query Store configuration from the SourceDatabase through the Get-DbaQueryStoreConfig function
+		$SourceQSConfig = Get-DbaQueryStoreConfig -SqlServer $sourceserver -Databases $SourceDatabase
+		
 	}
-
+	
 	PROCESS
 	{
-
-        Write-Verbose "Connecting to target: $TargetSqlServer"
-        try
+		if (!$DestinationDatabase -and !$exclude -and !$alldatabases)
 		{
-			$targetserver = Connect-SqlServer -SqlServer $TargetSqlServer -SqlCredential $Credential
-
+			Write-Warning "You must specify databases to execute against using either -DestinationDatabase, -Exclude or -AllDatabases"
+			continue
+		}
+		
+		Write-Verbose "Connecting to destination: $Destination"
+		try
+		{
+			$destserver = Connect-SqlServer -SqlServer $Destination -SqlCredential $Credential
```

Comment:
$Destination can accept multiple objects.
Probably it needs to be altered in the param definition?
If not, there needs to be a foreach to connect to multiple destinations.